Source/JavaScriptCore/dfg/DFGSpeculativeJIT64.cpp

```diff
@@ -6611,7 +6642,12 @@ void SpeculativeJIT::convertAnyInt(Edge valueEdge, GPRReg resultGPR, bool canIgn
     unboxDouble(valueGPR, resultGPR, scratch2FPR);
     branchConvertDoubleToInt52(scratch2FPR, resultGPR, failureCases, scratch1GPR, scratch3FPR, canIgnoreNegativeZero);
     done.link(this);
-    DFG_TYPE_CHECK(JSValueRegs(valueGPR), valueEdge, SpecInt32Only | SpecAnyIntAsDouble, failureCases);
+
+    auto filter = SpecInt32Only | SpecAnyIntAsDouble;
+    if (canIgnoreNegativeZero)
+        filter |= SpecDoubleReal;
+    m_interpreter.filter(valueEdge, filter);
+    speculationCheck(BadType, JSValueRegs(valueGPR), valueEdge.node(), failureCases);
```

Comment:
Can ww do this instead?
```
DFG_TYPE_CHECK(JSValueRegs(valueGPR), valueEdge, filter, failureCases);
```

Reply:
We can't quite do this, because `DFG_TYPE_CHECK` only links failure cases if the speculated type indicates that we need to filter. Here, we need to also perform range checks on the DoubleReal, so we need to ensure we link failure cases every time.